File: app/src/main/java/com/example/taxconnect/LoginActivity.java
package com.example.taxconnect;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.taxconnect.model.UserModel;
import com.google.android.material.textfield.TextInputEditText;

public class LoginActivity extends AppCompatActivity {

    private TextInputEditText etEmail, etPassword;
    private Button btnLogin;
    private TextView tvRegisterLink;
    private DataRepository repository;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        repository = DataRepository.getInstance();

        etEmail = findViewById(R.id.etEmail);
        etPassword = findViewById(R.id.etPassword);
        btnLogin = findViewById(R.id.btnLogin);
        tvRegisterLink = findViewById(R.id.tvRegisterLink);

        btnLogin.setOnClickListener(v -> loginUser());
        
        tvRegisterLink.setOnClickListener(v -> {
            // Default to Customer role registration if clicked from here, or go back to Main
            Intent intent = new Intent(LoginActivity.this, MainActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(intent);
        });
    }

    private void loginUser() {
        String email = etEmail.getText().toString().trim();
        String password = etPassword.getText().toString().trim();

        if (TextUtils.isEmpty(email) || TextUtils.isEmpty(password)) {
            Toast.makeText(this, "Please enter email and password", Toast.LENGTH_SHORT).show();
            return;
        }

        btnLogin.setEnabled(false);
        btnLogin.setText("Logging in...");

        repository.loginUser(email, password, new DataRepository.DataCallback<UserModel>() {
            @Override
            public void onSuccess(UserModel user) {
                Toast.makeText(LoginActivity.this, "Login Successful", Toast.LENGTH_SHORT).show();
                navigateToDashboard(user.getRole());
            }

            @Override
            public void onError(String error) {
                btnLogin.setEnabled(true);
                btnLogin.setText("Login");
                Toast.makeText(LoginActivity.this, "Error: " + error, Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void navigateToDashboard(String role) {
        Intent intent;
        if ("CA".equalsIgnoreCase(role)) {
            intent = new Intent(this, CADashboardActivity.class);
        } else {
            intent = new Intent(this, HomeActivity.class);
        }
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
    }
}

File: tax/TAX-connect-prototype/tax/app/src/main/java/com/example/taxconnect/features/auth/LoginActivity.kt
package com.example.taxconnect.features.auth

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Patterns
import android.view.LayoutInflater
import com.example.taxconnect.R
import com.example.taxconnect.core.base.BaseActivity
import com.example.taxconnect.data.repositories.DataRepository
import com.example.taxconnect.data.repositories.DataRepository.DataCallback
import com.example.taxconnect.databinding.ActivityLoginBinding
import com.example.taxconnect.data.models.UserModel
import com.example.taxconnect.features.dashboard.CADashboardActivity
import com.example.taxconnect.features.home.HomeActivity
import com.example.taxconnect.features.home.MainActivity
import com.example.taxconnect.core.network.NetworkUtils

class LoginActivity : BaseActivity<ActivityLoginBinding>() {

    override val bindingInflater: (LayoutInflater) -> ActivityLoginBinding = ActivityLoginBinding::inflate
    private lateinit var repository: DataRepository

    override fun initViews() {
        repository = DataRepository.getInstance()
    }

    override fun observeViewModel() {
        // No ViewModel to observe
    }

    override fun setupListeners() {
        binding.btnLogin.setOnClickListener { loginUser() }

        binding.tvRegisterLink.setOnClickListener {
            val intent = Intent(this@LoginActivity, MainActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            startActivity(intent)
        }
    }

    private fun loginUser() {
        if (!NetworkUtils.isNetworkAvailable(this)) {
            showToast(getString(R.string.no_internet))
            return
        }

        val email = binding.etEmail.text.toString().trim()
        val password = binding.etPassword.text.toString().trim()

        if (TextUtils.isEmpty(email)) {
            binding.etEmail.error = getString(R.string.email_required)
            binding.etEmail.requestFocus()
            return
        }

        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            binding.etEmail.error = getString(R.string.invalid_email)
            binding.etEmail.requestFocus()
            return
        }

        if (TextUtils.isEmpty(password)) {
            binding.etPassword.error = getString(R.string.password_required)
            binding.etPassword.requestFocus()
            return
        }

        if (password.length < 6) {
            binding.etPassword.error = getString(R.string.password_min_length)
            binding.etPassword.requestFocus()
            return
        }

        binding.btnLogin.isEnabled = false
        binding.btnLogin.text = getString(R.string.logging_in)

        repository.loginUser(email, password, object : DataCallback<UserModel> {
            override fun onSuccess(data: UserModel?) {
                showToast(getString(R.string.login_successful))
                if (data != null) {
                    navigateToDashboard(data.role)
                }
            }

            override fun onError(error: String?) {
                binding.btnLogin.isEnabled = true
                binding.btnLogin.text = getString(R.string.login)
                showToast(error ?: "Login failed")
            }
        })
    }

    private fun navigateToDashboard(role: String?) {
        val intent = if ("CA".equals(role, ignoreCase = true)) {
            Intent(this, CADashboardActivity::class.java)
        } else {
            Intent(this, HomeActivity::class.java)
        }
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        startActivity(intent)
    }
}
